src/sec_adapter_soc_provisioning.c

```diff
@@ -0,0 +1,879 @@
+/**
+ * Copyright 2020-2023 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdio.h>
+#include <sys/stat.h>
+#include <stdlib.h>
+#include "sec_adapter_key.h" // NOLINT
+#include "sa.h"
+#include "sec_adapter_cipher.h"
+#include "sec_adapter_key_legacy.h"
+#include "sec_adapter_processor.h"
+
+#define MAX_PATHS 12
+#define MAX_NAME_LEN 64
+#define MAX_PATH_LEN 128
+
+
+
+#define WIDEVINE_PRIVATE_KEY         0x0351000003510001  // OEM Private Key
+#define WIDEVINE_CERT_BUNDLE         0x0651000006510001  // Certificate Bundle
+
+#define PLAYREADY_2K_PRIVATE_KEY     0x0331000003310001  // 2K Model Private Key
+#define PLAYREADY_3K_PRIVATE_KEY     0x0331000003310003  // 3K Model Private Key
+#define PLAYREADY_2K_CERTIFICATE     0x0631000006310001  // 2K Model Certificate
+#define PLAYREADY_3K_CERTIFICATE     0x0631000006310002  // 3K Model Certificate
+
+#define PROV_NETFLIX_KDE                  0x0371000003710001
+#define PROV_NETFLIX_KDH                  0x0371000003710002  // Device HMAC Key
+#define PROV_NETFLIX_KDW                  0x0371000003710003  // Device Wrapping Key
+#define PROV_NETFLIX_ESN                  0x0671000006710001  // ESN Container
+#define PROV_NETFLIX_DRMBIN               0x0671000006710002  // DRM Binary
+
+#define MFI_BASE_KEY                 0x0391000003910001  // MFi Base Key
+#define MFI_PROVISIONING_OBJECT      0x0691000006910001  // MFi Provisioning Object
+
+#define FAIRPLAY_SECRET              0x0691000006910002  // FairPlay Secret
+
+
+#define BASE_PATH "/opt/drm.prov"
+typedef struct {
+    char description[MAX_NAME_LEN];
+    char filePath[MAX_PATH_LEN];
+} FilePath;
+
+FilePath resolvedWideVinePaths[] = {
+    {"Widevine OEM Private Key", "/opt/drm/0351000003510001.key"},
+    {"Widevine OEM Certificate", "/opt/drm/0651000006510001.bin"}
+};
+
+FilePath resolvedPathsPlayReady2K[] = {
+    {"PlayReady 2K OEM Private Key", "/opt/drm/0331000003310001.key"},
```

Comment:
I don't know if we can really use these full path names.  The "/opt/drm" path is set in the SocProcessorGetInstanceDirectory function.  Would it be possible to derive these path from the passed in directory and the value in sec_security_comcastids.h?